src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RTHttpClientHandler.cs

```diff
@@ -0,0 +1,475 @@
+using RTHttpCookieUsageBehavior = Windows.Web.Http.Filters.HttpCookieUsageBehavior;
+using RTPasswordCredential = Windows.Security.Credentials.PasswordCredential;
+
+namespace System.ServiceModel.Channels
+{
+    internal class RTHttpClientHandler : DelegatingHandler
+    {
+        private static readonly Lazy<bool> s_RTCookieUsageBehaviorSupported =
+            new Lazy<bool>(InitRTCookieUsageBehaviorSupported);
+
+        //#region Fields
+
+        private readonly RTHttpBaseProtocolFilter _rtFilter;
+        private readonly RTHttpHandlerToFilter _handlerToFilter;
+
+        private volatile bool _operationStarted;
+        private volatile bool _disposed;
+
+        private CookieContainer _cookieContainer;
+        private bool _useCookies;
+        private DecompressionMethods _automaticDecompression;
+        private X509Certificate2Collection _clientCertificates;
+
+
+        //#endregion Fields
+
+        //#region Properties
+
+        public bool UseCookies
+        {
+            get { return _useCookies; }
+            set
+            {
+                CheckDisposedOrStarted();
+                _useCookies = value;
+            }
+        }
+
+        public CookieContainer CookieContainer
+        {
+            get { return _cookieContainer; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+                if (!UseCookies)
+                {
+                    throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture,
+                        "net_http_invalid_enable_first", "UseCookies", "true"));
+                }
+                CheckDisposedOrStarted();
+                _cookieContainer = value;
+            }
+        }
+
+        public X509Certificate2Collection ClientCertificates
+        {
+            get { return _clientCertificates; }
+        }
+
+        public DecompressionMethods AutomaticDecompression
+        {
+            get { return _automaticDecompression; }
+            set
+            {
+                CheckDisposedOrStarted();
+
+                // Automatic decompression is implemented downstack.
+                // HBPF will decompress both gzip and deflate, we will set
+                // accept-encoding for one, the other, or both passed in here.
+                _rtFilter.AutomaticDecompression = (value != DecompressionMethods.None);
+                _automaticDecompression = value;
+            }
+        }
+
+        public bool PreAuthenticate
+        {
+            get { return true; }
+            set
+            {
+                if (value != PreAuthenticate)
+                {
+                    throw new PlatformNotSupportedException(String.Format(CultureInfo.InvariantCulture,
+                        "net_http_value_not_supported", value, "PreAuthenticate"));
```

Comment:
Similar kind of question about whether we need SR.Format and what kind of message this will produce.